### PR TITLE
docs: NodePort XDP on AWS

### PR DESCRIPTION
**Pull request by borkmann**
Write up a small getting started guide based on Tobias' notes on
EKS (https://gist.github.com/tklauser/268641aea4fced9e8c3b4a2f7536661b).

Signed-off-by: Tobias Klauser <tklauser@distanz.ch>
Signed-off-by: Daniel Borkmann <daniel@iogearbox.net>

### CODE REVIEW

File: Documentation/gettingstarted/kubeproxy-free.rst [hunk 1]
```diff
@@ -459,6 +459,119 @@ right at the XDP layer are not visible in tcpdump since packet taps come at a mu
...
+  for testing purposes.
+
+Before Cilium's XDP acceleration can be deployed, there are two settings needed on the
+network adapter side, that is, MTU needs to be lowered in order to be able to operate
+with XDP, and number of combined channels need to be adapted.
+
+The default MTU is set to 9001 on the ena driver. Given XDP buffers are linear, they
+operate on a single page. A driver typically reserves some headroom for XDP as well
+(e.g. for encapsulation purpose), therefore, the highest possible MTU for XDP would
+be 3818.
```
Comment (jrfastab) [inline Documentation/gettingstarted/kubeproxy-free.rst:526]: Limitation is a bit unfortunate. I wonder if we should add an option to configure mtu for 1.9?

File: Documentation/gettingstarted/kubeproxy-free.rst [hunk 2]
```diff
@@ -459,6 +459,119 @@ right at the XDP layer are not visible in tcpdump since packet taps come at a mu
...
+  Other:          0
+  Combined:       4
+  Current hardware settings:
+  RX:             0
+  TX:             0
+  Other:          0
+  Combined:       4
+
+In order to use XDP the channels must be set to at most 1/2 of the value from
+``Combined`` above. Both, MTU and channel changes are applied as follows:
```
Comment (jrfastab) [inline Documentation/gettingstarted/kubeproxy-free.rst:546]: iirc some drivers will automagically reconfigure to meet the rx/tx queue requirements perhaps we should file a feature request against ena to do this.

**Review by jrfastab — APPROVED** [review]
LGTM perhaps in future release we can make this a bit more seamless but this is awesome to see XDP being used here.

File: Documentation/gettingstarted/kubeproxy-free.rst (hunk 2 above)
Comment (borkmann) [reply Documentation/gettingstarted/kubeproxy-free.rst:546]: Agree, either that or implement it directly upstream. E.g. on Azure no additional knobs are needed.

File: Documentation/gettingstarted/kubeproxy-free.rst (hunk 1 above)
Comment (borkmann) [reply Documentation/gettingstarted/kubeproxy-free.rst:526]: Yes, the agent could take care of it for 1.9.

**Pull request merged**